### ISSUE TITLE
Help Wanted: Clean up tree creation in ambients.pegjs 

### ISSUE THREAD

**Issue by aphelionz** [ambientsprotocol/ambc#4]
Serial execution is modeled in the AST is via a linear tree structure by the child node, something like:

```json
{
  "type": "Serial",
  "children": [{
    "id": "a",
    "type": "In_",  
    "children": [{  
      "id": "b",    
      "type": "In_",
      "children": [{
        "id": "d",  
        "type": "In"
      }]            
    }]              
  }]            
}
```

This is handled in the pegjs file here: https://github.com/aphelionz/ambc/blob/master/src/ambients.pegjs#L18. While this works, it would be nice to have a more idiomatic way of doing it to make it more readable and clear what the action is.

**Comment by tabcat** [comment]
```js
tail.reduce((acc, cur, i) => [{ 
  ...tail[i],
  ...(tail.length - 1 > i ? { children: [cur] } : {})
}], [])
```
this *should* do the same thing @aphelionz

**Comment by tabcat** [comment]
```js
tail.reduce(( a, c, i) => [{
  ...(tail.length - 1 > i ? { ...c, children: [c] } : { ...c })
}], [])
```
this *should* work @aphelionz

**Comment by aphelionz** [comment]
Fixed in #5
